command/agentproxyshared/cache/lease_cache.go

```diff
@@ -644,27 +660,35 @@ func (c *LeaseCache) storeStaticSecretIndex(ctx context.Context, req *SendReques
 	// update the index with the new capability:
 	capabilitiesIndex.ReadablePaths[path] = struct{}{}
 
-	err = c.db.SetCapabilitiesIndex(capabilitiesIndex)
+	err = c.SetCapabilitiesIndex(ctx, capabilitiesIndex)
 	if err != nil {
 		c.logger.Error("failed to cache token capabilities as part of caching the proxied response", "error", err)
 		return err
 	}
 
+	// Lastly, ensure that we start renewing this index, if it's  new.
+	// We require the 'created' check so that we don't renew the same
+	// index multiple times.
+	if c.capabilityManager != nil && created {
+		c.capabilityManager.StartRenewingCapabilities(capabilitiesIndex)
```

Comment:
We use `created` to know if we should start renewing. Without this check, we'd be renewing the same set of capabilities multiple times.